### PR TITLE
Marketplace: Add category result text / title and clear link

### LINKED ISSUE #63791: Marketplace: Add category result title
The FullListView component needs similar title changes as seen in https://github.com/Automattic/wp-calypso/pull/63645

I think to implement this we should create a new CategoryListView component and rename FullListView to PaidPopularListView as that's all its used for. We still need this separate view because it shows the monthly / annual price selector.

- [ ] Before starting: Check design decision on https://github.com/Automattic/wp-calypso/issues/63790, there might be conflict on what we want to do here.
- [ ] On selecting a category, the title should update to match the style of the search result pages 

Currently:
Screenshot 2022-05-19 at 12-00-16 Plugins — WordPress com

Expect something like this:
Screenshot 2022-05-19 at 12-03-21 Plugins — WordPress com

### LINKED ISSUE #63839: Marketplace: No visual indicator for selected category for the category select on the mobile view
Screen Capture on 2022-05-19 at 18-22-30

On the swipable menu when we are in a category list if the selected category is on the middle of the list the user can't see a visual indicator of that category being selected until its scrolled there.

### PR DESCRIPTION
**Pull request by lsl**
#### Changes proposed in this Pull Request

* Adds "Found X plugins. Clear" text for categories similar to the text added for search results in https://github.com/Automattic/wp-calypso/pull/63645
* Cleans up css
* Removes excessive margins between search and category selector cc @SaxonF fyi in case this wasn't bug

#### Testing instructions

* http://calypso.localhost:3000/plugins
* Selecting a category should display the result count and a clear button
* Clear link should clear the selected category
* Note, on mobile clicking clear won't scroll category to the start to show discover selected again, this is fixed in https://github.com/Automattic/wp-calypso/pull/63842 (likely gets merged first anyway)

Before
Screenshot 2022-05-20 at 15-23-05 Plugins ‹ Site Title — WordPress com

After
Screenshot 2022-05-23 at 15-46-53 Plugins — WordPress com

Fixes https://github.com/Automattic/wp-calypso/issues/63839 
Fixes https://github.com/Automattic/wp-calypso/issues/63791

### CODE REVIEW

File: client/my-sites/plugins/categories/responsive-toolbar-group/dropdown-group.tsx [hunk 1]
```diff
@@ -193,6 +193,11 @@ export default function DropdownGroup( {
 		};
 	}, [ shadowListItems, interceptionCallback, hideRatio, showRatio, rootMargin ] );
 
+	// Reset active on prop change from above
+	useEffect( () => {
```
Comment (Rolilink) [inline client/my-sites/plugins/categories/responsive-toolbar-group/dropdown-group.tsx:197]: thanks for this one :)! I was thinking on improving the component in the future to make it more dynamic and less self-controlled and this one was on my notes.

**Review by Rolilink — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7329023

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7329023).

**Pull request merged**